### PR TITLE
Change unique identifier to (uid, owner, group, queryparams), was before (uid, owner, group)

### PR DESCRIPTION
**Pull request by ksuess**
The change to unique identifier (uid, owner, group, queryparams) allows to bookmark query urls of a faceted navigation.

### CODE REVIEW

**Review by jensens — APPROVED** [review]
overall LGTM
Does this affect the Svelte code in here?

**Comment by ksuess** [comment]
thought about a constructive answer about the Svelte implementation.

I would love to dig more in the Svelte code. But decided to concentrate on a REST API that is OK with the former identifier (UID, owner, group) as my changes handle the additional part of the identifier as **optional**. 
Why didn't I explore the Svelte implementation?
First impression:
localStorage
Using localStorage with a key "collective.bookmarks" means that 
* all sites with collective.bookmarks are sharing their bookmarks as they share this storage
* testing is quite difficult as soon as you have more than one installation.
localStorage.clear(); helps for testing, but, well.

Please commit the code for the viewlet for a button to bookmark a page. That would help testing the Svelte implementation.

Be careful to work on master as it is out of date. Branch 'mrtango-cusom-elements' is THE branch for now.

**Comment by jensens** [comment]
@agitator please eye on this, because you're using it ;)

**Comment by ksuess** [comment]
@agitator  Your add-on provides Svelte code for toggling bookmark and displaying bookmarks to place in viewlets or portlets. Do you want to add such viewlets or an instruction how your intention of integratiing the Svelte code should be done, or can I? That would help testing if my changes (just requesting Rest API for Volto bookmarks, no integration of Svelte code) are OK with the current state.

File: src/collective/bookmarks/api/utils.py [hunk 1]
```diff
@@ -37,10 +37,20 @@ def get_bookmark_from_request(request, loadjson=False):
...
     group = data.get("group", "")
     owner = get_owner(data=data)
-    return owner, uid, group, data.get("payload", {})
+    return owner, uid, group, data.get("queryparams", ''), data.get("payload", {})
 
 
 def bookmark_dict_to_json_dict(bookmark):
-    bookmark["@type"] = "collective.bookmarks.bookmark"
-    bookmark["uid"] = str(bookmark["uid"])
+    """Prepare result item for responseself.
```
Comment (jensens) [inline src/collective/bookmarks/api/utils.py]: typo?

**Review by agitator — APPROVED** [review]
works with my existing project

**Pull request merged**